From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bacteriawa <A3167717663@hotmail.com>
Date: Fri, 13 Feb 2026 16:17:36 +0800
Subject: [PATCH] Leaves: Jade Protocol

This patch is Powered by Jade(https://github.com/Snownee/Jade)
Co-authored by: violetc <58360096+s-yh-china@users.noreply.github.com>
As a part of : Leaves (https://github.com/LeavesMC/Leaves)
Licensed under: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

diff --git a/net/minecraft/world/entity/animal/armadillo/Armadillo.java b/net/minecraft/world/entity/animal/armadillo/Armadillo.java
index 41fd33b6d70eacd987862915b10e9b559867f44d..ccfd8993a64d279be9f4339a03cda717d1a5987e 100644
--- a/net/minecraft/world/entity/animal/armadillo/Armadillo.java
+++ b/net/minecraft/world/entity/animal/armadillo/Armadillo.java
@@ -75,7 +75,7 @@ public class Armadillo extends Animal {
     public final AnimationState rollOutAnimationState = new AnimationState();
     public final AnimationState rollUpAnimationState = new AnimationState();
     public final AnimationState peekAnimationState = new AnimationState();
-    private int scuteTime;
+    public int scuteTime; // Leaves - private -> public
     private boolean peekReceivedClient = false;
 
     public Armadillo(final EntityType<? extends Animal> type, final Level level) {
diff --git a/net/minecraft/world/entity/animal/frog/Tadpole.java b/net/minecraft/world/entity/animal/frog/Tadpole.java
index 296cae1d5f5a3a054949ee03e2a2552bfeae4682..97c49875269dd52c6a4692940b122f1d1f3fd2fb 100644
--- a/net/minecraft/world/entity/animal/frog/Tadpole.java
+++ b/net/minecraft/world/entity/animal/frog/Tadpole.java
@@ -256,7 +256,7 @@ public class Tadpole extends AbstractFish {
         }
     }
 
-    private int getTicksLeftUntilAdult() {
+    public int getTicksLeftUntilAdult() { // Leaves - private -> public
         return Math.max(0, ticksToBeFrog - this.age);
     }
 
diff --git a/net/minecraft/world/level/storage/loot/LootPool.java b/net/minecraft/world/level/storage/loot/LootPool.java
index 475e00df0cbbeb2dcd682ad03ad42320d89c762b..c79c39b01b3cbfc6b9b7d2c000182cc16d11cee9 100644
--- a/net/minecraft/world/level/storage/loot/LootPool.java
+++ b/net/minecraft/world/level/storage/loot/LootPool.java
@@ -37,7 +37,7 @@ public class LootPool implements Validatable {
             )
             .apply(i, LootPool::new)
     );
-    private final List<LootPoolEntryContainer> entries;
+    public final List<LootPoolEntryContainer> entries; // Leaves - private -> public
     private final Optional<Holder<LootItemCondition>> condition;
     private final Optional<Holder<LootItemFunction>> modifier;
     private final Holder<ContextIntProvider> rolls;
diff --git a/net/minecraft/world/level/storage/loot/LootTable.java b/net/minecraft/world/level/storage/loot/LootTable.java
index 10c384c5c828115f361f6e3b2c4114ba167a027f..3498420b6406c236fa011be83a86b909d0ec7de8 100644
--- a/net/minecraft/world/level/storage/loot/LootTable.java
+++ b/net/minecraft/world/level/storage/loot/LootTable.java
@@ -50,7 +50,7 @@ public class LootTable implements Validatable {
     public static final LootTable EMPTY = new LootTable(LootContextParamSets.EMPTY, Optional.empty(), List.of(), Optional.empty());
     private final ContextKeySet paramSet;
     private final Optional<Identifier> randomSequence;
-    private final List<LootPool> pools;
+    public final List<LootPool> pools; // Leaves - private -> public
     private final Optional<Holder<LootItemFunction>> modifier;
     public org.bukkit.craftbukkit.CraftLootTable craftLootTable; // CraftBukkit
 
diff --git a/net/minecraft/world/level/storage/loot/entries/CompositeEntryBase.java b/net/minecraft/world/level/storage/loot/entries/CompositeEntryBase.java
index 898b0d1238296bbd325059fac21b322085f2804b..457e607645a8a4b0d94be20714713b1c94aec27d 100644
--- a/net/minecraft/world/level/storage/loot/entries/CompositeEntryBase.java
+++ b/net/minecraft/world/level/storage/loot/entries/CompositeEntryBase.java
@@ -21,7 +21,7 @@ public abstract class CompositeEntryBase extends LootPoolEntryContainer {
             return "Empty children list";
         }
     };
-    protected final List<LootPoolEntryContainer> children;
+    public final List<LootPoolEntryContainer> children; // Leaves - private -> public
     private final ComposableEntryContainer composedChildren;
 
     protected CompositeEntryBase(
diff --git a/net/minecraft/world/level/storage/loot/entries/LootPoolEntryContainer.java b/net/minecraft/world/level/storage/loot/entries/LootPoolEntryContainer.java
index 08a00034b3b3b26638cd18fe28d3ecbc78bbc4f2..18721bbda9622353d6bb0beada9b1b623cc9c374 100644
--- a/net/minecraft/world/level/storage/loot/entries/LootPoolEntryContainer.java
+++ b/net/minecraft/world/level/storage/loot/entries/LootPoolEntryContainer.java
@@ -19,7 +19,7 @@ import net.minecraft.world.level.storage.loot.predicates.ConditionUserBuilder;
 import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
 
 public abstract class LootPoolEntryContainer implements ComposableEntryContainer, Validatable {
-    protected final Optional<Holder<LootItemCondition>> condition;
+    public final Optional<Holder<LootItemCondition>> condition; // Leaves - private -> public
     protected final Optional<Holder<LootItemFunction>> modifier;
 
     protected LootPoolEntryContainer(final Optional<Holder<LootItemCondition>> condition, final Optional<Holder<LootItemFunction>> modifier) {
diff --git a/net/minecraft/world/level/storage/loot/entries/NestedLootTable.java b/net/minecraft/world/level/storage/loot/entries/NestedLootTable.java
index dbd606f430ec8e33b1b30f4d3ecc65c2d3313782..30b0ff5354b3186fd350e76688ee316b35e5396e 100644
--- a/net/minecraft/world/level/storage/loot/entries/NestedLootTable.java
+++ b/net/minecraft/world/level/storage/loot/entries/NestedLootTable.java
@@ -18,7 +18,7 @@ public class NestedLootTable extends ExpandableContainerBase {
     public static final MapCodec<NestedLootTable> MAP_CODEC = RecordCodecBuilder.mapCodec(
         i -> i.group(LootTable.LIST_CODEC.fieldOf("value").forGetter(e -> e.value)).and(expandableFields(i)).apply(i, NestedLootTable::new)
     );
-    private final HolderSet<LootTable> value;
+    public final HolderSet<LootTable> value; // Leaves - private -> public
 
     private NestedLootTable(
         final HolderSet<LootTable> value,
diff --git a/net/minecraft/world/level/storage/loot/predicates/CompositeLootItemCondition.java b/net/minecraft/world/level/storage/loot/predicates/CompositeLootItemCondition.java
index 4999efdbf7972590d6f783ee455605bb91735a10..70f959309d28a326271b6300b3a08373ccd1f994 100644
--- a/net/minecraft/world/level/storage/loot/predicates/CompositeLootItemCondition.java
+++ b/net/minecraft/world/level/storage/loot/predicates/CompositeLootItemCondition.java
@@ -13,7 +13,7 @@ import net.minecraft.world.level.storage.loot.Validatable;
 import net.minecraft.world.level.storage.loot.ValidationContext;
 
 public abstract class CompositeLootItemCondition implements LootItemCondition {
-    protected final HolderSet<LootItemCondition> terms;
+    public final HolderSet<LootItemCondition> terms; // Leaves - private -> public
     private final Predicate<LootContext> composedPredicate;
 
     protected CompositeLootItemCondition(final HolderSet<LootItemCondition> terms, final Predicate<LootContext> composedPredicate) {
